File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/Paging.java
package work.socialhub.kslack.entity;

public class Paging {

    private String iid; // search
    private Integer count;
    private Integer total;
    private Integer page;
    private Integer pages;
    private Integer perPage;
    private Integer spill;

    public String getIid() {
        return this.iid;
    }

    public Integer getCount() {
        return this.count;
    }

    public Integer getTotal() {
        return this.total;
    }

    public Integer getPage() {
        return this.page;
    }

    public Integer getPages() {
        return this.pages;
    }

    public Integer getPerPage() {
        return this.perPage;
    }

    public Integer getSpill() {
        return this.spill;
    }

    public void setIid(String iid) {
        this.iid = iid;
    }

    public void setCount(Integer count) {
        this.count = count;
    }

    public void setTotal(Integer total) {
        this.total = total;
    }

    public void setPage(Integer page) {
        this.page = page;
    }

    public void setPages(Integer pages) {
        this.pages = pages;
    }

    public void setPerPage(Integer perPage) {
        this.perPage = perPage;
    }

    public void setSpill(Integer spill) {
        this.spill = spill;
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/Paging.kt
package work.socialhub.kslack.entity

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.js.JsExport

/**
 * Represents pagination metadata in Slack API responses.
 *
 * Contains page number, total count, per-page count, and total pages
 * for paginated results. Used in search and list API responses.
 * The `iid` field is specific to search results.
 */
@JsExport
@Serializable
class Paging {

    var iid: String? = null // search
    var count: Int? = null
    var total: Int? = null
    var page: Int? = null
    var pages: Int? = null

    @SerialName("per_page")
    var perPage: Int? = null

    var spill: Int? = null
}
